tests/components/lastfm/test_config_flow.py

```diff
@@ -0,0 +1,289 @@
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+        )
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input=CONF_USER_DATA,
+        )
+        assert result["type"] == data_entry_flow.FlowResultType.FORM
+        assert result["step_id"] == "friends"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=CONF_FRIENDS_DATA
+        )
+        assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
+        assert result["title"] == DEFAULT_NAME
+        assert result["options"] == CONF_DATA
+
+
+async def test_flow_user_invalid_auth(hass: HomeAssistant) -> None:
+    """Test user initialized flow with invalid authentication."""
+    with patch_fetch_user(
+        thrown_error=WSError(
+            "network",
+            "status",
+            "Invalid API key - You must be granted a valid key by last.fm",
+        )
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CONF_USER_DATA
+        )
+        assert result["type"] == data_entry_flow.FlowResultType.FORM
+        assert result["step_id"] == "user"
+        assert result["errors"]["base"] == "invalid_auth"
+
+
+async def test_flow_user_invalid_username(hass: HomeAssistant) -> None:
+    """Test user initialized flow with invalid username."""
+    with patch_fetch_user(thrown_error=WSError("network", "status", "User not found")):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CONF_USER_DATA
+        )
+        assert result["type"] == data_entry_flow.FlowResultType.FORM
+        assert result["step_id"] == "user"
+        assert result["errors"]["base"] == "invalid_account"
+
+
+async def test_flow_user_unknown(hass: HomeAssistant) -> None:
+    """Test user initialized flow with unknown error."""
+    with patch_fetch_user(thrown_error=Exception()):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CONF_USER_DATA
+        )
+        assert result["type"] == data_entry_flow.FlowResultType.FORM
+        assert result["step_id"] == "user"
+        assert result["errors"]["base"] == "unknown"
+
+
+async def test_flow_user_unknown_lastfm(hass: HomeAssistant) -> None:
+    """Test user initialized flow with unknown error from lastfm."""
+    with patch_fetch_user(
+        thrown_error=WSError("network", "status", "Something strange")
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CONF_USER_DATA
+        )
+        assert result["type"] == data_entry_flow.FlowResultType.FORM
+        assert result["step_id"] == "user"
+        assert result["errors"]["base"] == "unknown"
```

Comment:
Combine these tests into one for error after error and end test result with a success (to ensure config flow recovers from errors)